codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/PaginatorGenerator.kt

```diff
@@ -48,13 +48,15 @@ class PaginatorGenerator private constructor(
     service: ServiceShape,
     operation: OperationShape,
     private val generics: FluentClientGenerics,
+    retryPolicyType: Any = RustType.Unit,
```

Comment:
I'd make this a union type if I could but Kotlin doesn't support those and I couldn't find another way to say that it could be a `Symbol`, `RuntimeType`, `RustType`, `String`, or `GenericsGenerator`.

If y'all can think of a nice solution to that, I'll update the PR. The biggest thing is that I want to use the same `Templateable` type everywhere we template stuff and I can overload some `fun`s that only take one arg but other `fun`s take multiple args for writable things in no set order like my `rustTypeParameters` function and I don't want to change that to no longer be variadic e.g. I want to support this use case:

```kotlin
rustTemplate(
    "some_fn::<#{type_params:W}>();",
    "type_params" to rustTypeParameters(
        symbolProvider.toSymbol(operation),
        RustType.Unit,
        runtimeConfig.smithyHttp().member("body::SdkBody"),
        GenericsGenerator(GenericTypeArg("A"), GenericTypeArg("B")),
    )
)
```

Reply:
For this specific arg, it's going to be a `RuntimeType` for all AWS clients and it's going to be `RustType.Unit` for "native" smithy clients.

Reply:
a couple options: 
- take `Writeable` as an argument and make the caller deal with it
- make unit a `RuntimeType`: `val Unit = RuntimeType(namespace = "()")`
- make a type alias `Formattable = Any` but is clearly documented